src/client/datascience/progress/kernelProgressReporter.ts

```diff
@@ -186,28 +202,38 @@ export class KernelProgressReporter implements IExtensionSyncActivationService {
                 return;
             }
             shownOnce = true;
-            void window.withProgress({ location: ProgressLocation.Notification, title }, async (progress) => {
-                const info = KernelProgressReporter.instance!.kernelResourceProgressReporter.get(key);
-                if (!info) {
-                    return;
-                }
-                info.reporter = progress;
-                // If we have any messages, then report them.
-                while (info.pendingProgress.length > 0) {
-                    const message = info.pendingProgress.shift();
-                    if (message === title) {
-                        info.progressList.push(message);
-                    } else if (message !== title && message) {
-                        info.progressList.push(message);
-                        progress.report({ message });
+            void window.withProgress(
+                { location: ProgressLocation.Notification, title },
+                async (progress, token: CancellationToken) => {
+                    const info = KernelProgressReporter.instance!.kernelResourceProgressReporter.get(key);
+                    if (!info) {
+                        return;
                     }
+                    info.reporter = progress;
+                    // If we have any messages, then report them.
+                    while (info.pendingProgress.length > 0) {
+                        const message = info.pendingProgress.shift();
+                        if (message === title) {
+                            info.progressList.push(message);
+                        } else if (message !== title && message) {
+                            info.progressList.push(message);
+                            progress.report({ message });
+                        }
+                    }
+                    await Promise.race([
+                        createPromiseFromCancellation({ token, cancelAction: 'resolve', defaultValue: true }),
+                        deferred.promise
+                    ]);
+                    if (token.isCancellationRequested) {
+                        info.tokenSources.forEach((t) => t.cancel());
```

Comment:
This is where the tokenSources are all canceled. This didn't used to do anything.

Reply:
Yeah I was using this in the install PR to allow canceling of the install of a module. Perhaps I should create a new interface for that instead. 

I'll back out the tokenSource changes for now.